### DISCUSSION TITLE
`Pick grep` with smart case?

### DISCUSSION THREAD

**Discussion by fannheyward** [nvim-mini/mini.nvim#558] [Q&A] [answered]
How to make `Pick grep` and `Pick grep_live` search with smart case?

Thank you guys!

**Comment by echasnovski** [comment, marked as answer]
Both `grep` and `grep_live` try to use available CLI tool but only with minimal set of arguments. In order to customize it, [use approach for the CLI tool in use](https://github.com/echasnovski/mini.nvim/blob/8e18eb9ba849d737acac1873cd78fc8058c1700a/doc/mini-pick.txt#L79-L84).

By default `rg` is used. In order to make it search with smart case, follow these steps:
- Create and start using [configuration file](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#configuration-file). Basically, create a file anywhere on the disk and export `RIPGREP_CONFIG_PATH` in shell config file.
- Add `--smart-case` line to the config file.

If you are using `git` or fallback tool, then I am afraid it is not possible to make it smart case. The `git grep` has only `--ignore-case` flag and fallback has none.

Side note: `grep` uses CLI tool to find initial matches for the `pattern`, afterwards it uses `default_match` from 'mini.pick' to match those matches. The latter support smart case by setting both 'ignorecase' and 'smartcase' options to `true` (or 1 in Vimscript).
